internal/db/message.go

```diff
@@ -331,18 +336,21 @@ func GetMessageFlags(ctx context.Context, client *ent.Client, messageIDs []imap.
 			return nil, err
 		}
 
-		messages = append(messages, chunkMessages...)
-	}
+		for _, msg := range chunkMessages {
+			mfs := MessageFlagSet{
+				ID:      msg.ID,
+				FlagSet: imap.NewFlagSet(),
+			}
 
-	curFlags := make(map[imap.InternalMessageID]imap.FlagSet, len(messageIDs))
+			for _, v := range msg.Edges.Flags {
+				mfs.FlagSet.AddToSelf(v.Value)
+			}
 
-	for _, message := range messages {
-		curFlags[message.ID] = imap.NewFlagSetFromSlice(xslices.Map(message.Edges.Flags, func(flag *ent.MessageFlag) string {
-			return flag.Value
-		}))
+			result = append(result, mfs)
+		}
```

Comment:
the `AddSelf` operations here would probably do a lot of unnecessary appends and reallocations; can you collect the msg.Edges.Flags values with one `xslices.Map` call and `AddSelf(flags...)` in one go?

Reply:
I'm avoiding the the allocation of a temporary slice here, since map returns a new slice.